code/analyticsedge/src/phone/java/com/adobe/marketing/mobile/AnalyticsExtension.java

```diff
@@ -48,6 +49,21 @@
     protected AnalyticsExtension(final ExtensionApi extensionApi) {
         super(extensionApi);
         registerEventListeners(extensionApi);
+        platformServices = new AndroidPlatformServices();
+        systemInfoService = platformServices.getSystemInfoService();
+    }
+
+    /**
+     * Constructor for testing purposes.
+     *
+     * @param extensionApi  {@link ExtensionApi} instance
+     * @param systemInfoService the mocked {@link SystemInfoService} instance
+     */
+    AnalyticsExtension(final ExtensionApi extensionApi, final SystemInfoService systemInfoService) {
```

Comment:
In another PR I have created a constructor for testing purpose, that takes platformservices as an argument.
Can you change the parameter systeminforservice to platformservice and you mock platform services to return mocked SystemInfoService?
In that case we will end up adding one constructor for testing purpose.

Reply:
sure i can sync this up